packages/ListBox/src/components/Options/helpers/options.js

```diff
@@ -1,7 +1,98 @@
+import "@paprika/helpers/lib/polyfills/elementClosest";
 import useListBox from "../../../useListBox";
 import invokeOnChange from "../../../helpers/invokeOnChange";
 import { KEY_PRESS, CLICK } from "../../../types";
 
+function getOptions(event) {
+  let HTMLNodeList = null;
+  if (event.target.dataset?.pkaAnchor === "list-filter-input") {
+    // the event is coming from the filter
+    HTMLNodeList = event.target
+      .closest("[data-pka-anchor='list-box-box']")
+      .querySelector("[data-pka-anchor='styled-list']");
+  } else {
+    HTMLNodeList = event.target.querySelector("[data-pka-anchor='styled-list']");
+  }
+
+  // filtering only results with role="option" attribute
+  return [...HTMLNodeList.children].filter(li => li.hasAttribute("role"));
```

Comment:
Could do `li.getAttribute("role") === "option"` for a more explicit expression that probably wouldn't need the comment. 🤷‍♂️

Reply:
I think is ok :)